File: Podstawy Programowania/C/L4/4.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

int main() {
	double tablica[200] = {0};
	int n = 0;
	char plik_dane[] = "wzrost.txt";
	
	while (true) {
		
		puts("Wybierz dzialanie matematyczne");
		puts("1. Wczytaj i wyświetl dane");
		puts("2. Oblicz i wyświetl wartość średnią");
		puts("0. Zakoncz");
		int tryb;
		scanf("%d", &tryb);
		
		if (tryb == 0) break;
		if (tryb > 2 || tryb <0) {
			puts("Nieprawidlowa operacja");
			continue;
		}
		

		switch (tryb) {
			case 1: {
				
		
				while (true) {
					puts("Wybierz liczbe elementow od 52 do 200");
					scanf("%d", &n);
				
					if (n < 52 || n > 200) {
						puts("Nieprawidlowa liczba");
						continue;
					}
					
					break;
				}
		
				FILE *plik = fopen (plik_dane, "r");
				if (!plik) {
					puts ("Nie moge czytac z pliku wyniki.txt");
					return 1;
				}
				
				for (int i = 0; i < n; i++) {
					fscanf(plik, "%lf", &tablica[i]);
				}
				
				for (int i = 0; i < n; i++) {
					printf("%.4lf ", tablica[i]);
					if ((i + 1) % 4 == 0) printf("\n");
				}
				
				fclose(plik);
				break;
			}
				
			case 2: {
				double srednia = 0;
				
				for (int i = 0; i < n; i++) {
					srednia += tablica[i];
				}
				
				srednia /= n;
				
				
				printf("Zapisac do pliku czy wyswietlic na ekranie. \nWpisz 1 jesli zapisac, aby wyswietlic nacisnij inny przycisk.\n");
				int czyzapis;
				
				scanf("%d", &czyzapis);
				
				if (czyzapis == 1) {
					FILE *plik = fopen ("srednia.txt", "a");
					
					fprintf(plik, "%s %lf %d \n", plik_dane, srednia, n);
					fclose(plik);
				
				} else {
					printf("Srednia pomiarow wzrostu dla %d pomiarow wynosi %lf\n", n, srednia);
				}
				break;
			}
		}
	}
	return 0;
}

File: Podstawy Programowania/Go/L4/4.go
package main

import (
	"fmt"
	"os"
)

func main() {
	tablica := [200]float64{0}
	n := 0
	plik_dane := "wzrost.txt"

	for true {

		fmt.Println("Wybierz dzialanie matematyczne")
		fmt.Println("1. Wczytaj i wyświetl dane")
		fmt.Println("2. Oblicz i wyświetl wartość średnią")
		fmt.Println("0. Zakoncz")
		var tryb int
		fmt.Scan(&tryb)

		if tryb == 0 {
			break
		}
		if tryb > 2 || tryb < 0 {
			fmt.Println("Nieprawidlowa operacja")
			continue
		}

		switch tryb {
		case 1:
			{
			elements:
				fmt.Println("Wybierz liczbe elementow od 52 do 200")
				fmt.Scan(&n)

				if n < 52 || n > 200 {
					fmt.Println("Nieprawidlowa liczba")
					goto elements
				}

				plik, err := os.Open(plik_dane)
				if err != nil {
					panic(err)
				}

				defer plik.Close()

				for i := 0; i < n; i++ {
					fmt.Fscanf(plik, "%f", &tablica[i])
				}

				for i := 0; i < n; i++ {
					fmt.Printf("%.4f ", tablica[i])
					if (i+1)%4 == 0 {
						fmt.Printf("\n")
					}
				}

				break
			}

		case 2:
			{
				srednia := 0.0

				for i := 0; i < n; i++ {
					srednia += tablica[i]
				}

				srednia /= float64(n)

				fmt.Printf("Zapisac do pliku czy wyswietlic na ekranie. \nWpisz 1 jesli zapisac, aby wyswietlic nacisnij inny przycisk.\n")
				var czyzapis int
				fmt.Scan(&czyzapis)

				if czyzapis == 1 {
					plik, err := os.Create("srednia.txt")
					if err != nil {
						panic(err)
					}
					fmt.Fprintf(plik, "%s %f %d \n", plik_dane, srednia, n)

				} else {
					fmt.Printf("Srednia pomiarow wzrostu dla %d pomiarow wynosi %f\n", n, srednia)
				}
				break
			}
		}
	}
}
